modal-api.md

```diff
@@ -0,0 +1,58 @@
+# Design ideas for modal dialogs
+
+## The simple approach
+
+Several of the UX modal designs are pretty simple, consisting of a title, some text and some buttons (examples). These prompt-like intefaces could be supported with a cross-frame message with the format:
+
+```TypeScript
+interface Prompt = {
+  title : string,
```

Comment:
This is getting into implementation details, but will want the modal requestor to have a few more options for behavior:

closable, clickOutsideToClose.

The host app can use these properties and it's own styles to decide if the model has a glasspane behind it and we can handle iframe blur events to support click outside.